dps-common/caasauth/caasauth.go

```diff
@@ -0,0 +1,254 @@
+
+	caasPath := fmt.Sprintf("https://%[1]s/oidc/authorize?scope=openid dps offline_access&response_type=code&client_id=6eda2bf3-95ce-499b-8e27-1c19c93bae12&redirect_uri=https://%[1]s/caas/cli&state=random-state&nonce=random-nonce", address+basePath)
+	u, e := url.Parse(caasPath)
+	if e != nil {
+		utils.LogErrorAndExit(e)
+	}
+	u.RawQuery = u.Query().Encode()
+	reader := bufio.NewReader(os.Stdin)
+	fmt.Println()
+	fmt.Println(u)
+	fmt.Println()
+	fmt.Print("Enter security code: ")
+	deviceCode, _ := reader.ReadString('\n')
+	deviceCode = strings.TrimSuffix(deviceCode, "\n")
+	tokenReq := newTokenRequest(deviceCode)
+
+	cbTransport := &Transport{client.New(address, basePath+baseApiPath, []string{"https"})}
+
+	tokens, err := getCaasToken("https://"+address+basePath+"/oidc/token", tokenReq)
+	if err != nil {
+		utils.LogErrorAndExit(err)
+	}
+	cbTransport.Runtime.DefaultAuthentication = client.BearerToken(tokens.AccessToken)
+	cbTransport.Runtime.Transport = LoggedTransportConfig
+	return cbTransport, tokens
+}
+
+func getCaasToken(identityUrl string, tokenReq interface{}) (*TokenResponse, error) {
+	reqBody, _ := json.Marshal(tokenReq)
+	req, err := http.NewRequest("POST", identityUrl, bytes.NewReader(reqBody))
+	if err != nil {
+		return nil, err
+	}
+	//req.Header.Add("Accept", "application/json")
+	req.Header.Add("Content-Type", "application/json")
+
+	c := &http.Client{
+		Transport: LoggedTransportConfig,
+	}
+
+	resp, err := c.Do(req)
+
+	if resp == nil && err == nil {
+		return nil, errors.New(fmt.Sprintf("Unknown error while connnecting to %s", identityUrl))
+	}
+
+	if resp == nil || resp.StatusCode >= 400 {
+		if err != nil {
+			return nil, err
+		}
+		return nil, errors.New(fmt.Sprintf("Error while connnecting to %s, please check your username and password or use flags for each command. (%s)", identityUrl, resp.Status))
+	}
+
+	var tokenResp TokenResponse
+	body, _ := ioutil.ReadAll(resp.Body)
+	err = json.Unmarshal(body, &tokenResp)
+
+	return &tokenResp, nil
+}
+
+func cutAndTrimAddress(address string) (string, string) {
+	for _, v := range PREFIX_TRIM {
+		address = strings.TrimPrefix(address, v)
+	}
+	address = strings.TrimRight(address, "/ ")
+	basePath := ""
+	slashIndex := strings.Index(address, "/")
+	if slashIndex != -1 {
+		basePath = address[slashIndex:]
+		address = address[0:slashIndex]
+	}
+	return address, basePath
+}
+
+type Transport struct {
+	Runtime *client.Runtime
+}
+
+func (t *Transport) Submit(operation *runtime.ClientOperation) (interface{}, error) {
+	operation.Reader = &noContentSafeResponseReader{OriginalReader: operation.Reader}
+	response, err := t.Runtime.Submit(operation)
+	return response, err
+}
+
+type noContentSafeResponseReader struct {
+	OriginalReader runtime.ClientResponseReader
+}
+
+type ErrorMessage struct {
+	Message         string            `json:"message"`
+	ValidationError map[string]string `json:"validationErrors"`
+}
+
+func (e *ErrorMessage) String() string {
```

Comment:
Method needs more go style :)
```
	if len(e.Message) > 0 {
		return e.Message
	}
	if len(e.ValidationError) > 0 {
		var validationErrors []string
		for k, v := range e.ValidationError {
			message := fmt.Sprintf("'%v' - %v", strings.TrimPrefix(k, "postPrivate.arg0."), v)
			validationErrors = append(validationErrors, message)
		}
		return strings.Join(validationErrors, ",")
	}
	return ""
```

Reply:
I can't really see the difference

Reply:
The different is the style, and the suggested one follows Go standards, the other one is written in a different language :D